fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContextConfig.java

```diff
@@ -0,0 +1,308 @@
+/*
+ * FDBRecordContextConfig.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Map;
+
+/**
+ * A configuration struct that can be used to set various options on an {@link FDBRecordContext}. Instances
+ * of this configuration object can be passed to {@link FDBDatabase#openContext(FDBRecordContextConfig)}
+ * to create a new transaction with various parameters set according to the values specified here.
+ */
+public class FDBRecordContextConfig {
```

Comment:
(just a comment, nothing to do) I kind of have a preference for `FDBRecordContext.Config`, but given the size of this class I get why you did this.

Reply:
Yeah, that was my logic, too. Plus, as this would have a builder, it seemed that that would mean we either needed a `FDBRecordContext.Config.Builder` or a `FDBRecordContext.ConfigBuilder`, neither of which I was particularly excited about.